internal/controller/pdbwatcher_controller.go

```diff
@@ -3,311 +3,117 @@ package controllers
-		pdbWatcher.Status.TargetGeneration = target.Obj().GetGeneration()
-		pdbWatcher.Status.LastEviction = pdbWatcher.Spec.LastEviction //we could still keep a log here if thats useful
-		//should we clear evictions?
-		err = r.Status().Update(ctx, pdbWatcher)
-		if err != nil {
-			logger.Error(err, "Failed to update PDBWatcher status")
-			return ctrl.Result{}, err
-		}
-
-		// Log the scaling action
-		logger.Info(fmt.Sprintf("Scaled up Deployment %s/%s to %d replicas", target.Obj().GetNamespace(), target.Obj().GetName(), newReplicas))
-		return ctrl.Result{}, nil
-
-	}
-
-	// Watch for changes in PDB to revert to original state
-	if target.GetReplicas() != pdbWatcher.Status.MinReplicas {
-
-		// Revert Target to the original state
-		// This is bad if its a statefulset because scale down removes non zero replicas first even if zero isn't ready
-		target.SetReplicas(pdbWatcher.Status.MinReplicas)
-		err = r.Update(ctx, target.Obj())
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-
-		// Log the scaling action
-		logger.Info(fmt.Sprintf("Reverted Deployment %s/%s to %d replicas", target.Obj().GetNamespace(), target.Obj().GetName(), target.GetReplicas()))
-
-		// Update ResourceVersion in PDBWatcher status
-		pdbWatcher.Status.TargetGeneration = target.Obj().GetGeneration()
-		err = r.Status().Update(ctx, pdbWatcher)
-		if err != nil {
-			logger.Error(err, "Failed to update PDBWatcher status")
-			return ctrl.Result{}, err
+			log.Log.Info("Deleted PDBWatcher", "name", req.NamespacedName)
 		}
+		return reconcile.Result{}, nil
 	}
 
-	return ctrl.Result{}, nil
-}
-
-// does this go away if we force the pdb watcher name to be same as pdb?
-func (r *PDBWatcherReconciler) conflicts(ctx context.Context, pdbWatcher *myappsv1.PDBWatcher) error {
-	// Check for conflicts with other PDBWatchers
-	conflictWatcherList := &myappsv1.PDBWatcherList{}
-	err := r.List(ctx, conflictWatcherList, &client.ListOptions{Namespace: pdbWatcher.Namespace})
+	// If the PDB exists, create a corresponding PDBWatcher if it does not exist
+	var pdbWatcher PDBWatcher
+	err = r.Get(ctx, req.NamespacedName, &pdbWatcher)
 	if err != nil {
-		return err // Error listing PDBWatchers
-	}
-
-	for _, watcher := range conflictWatcherList.Items {
-		if watcher.Name != pdbWatcher.Name && watcher.Spec.PDBName == pdbWatcher.Spec.PDBName {
-			// Conflict detected
-			err := fmt.Errorf("PDB %s is already being watched by another PDBWatcher %s", pdbWatcher.Spec.PDBName, watcher.Name)
-			log.FromContext(ctx).Error(err, "conflict!")
-			r.Recorder.Event(pdbWatcher, corev1.EventTypeWarning, "Conflict", err.Error())
-			return err
+		// Create a new PDBWatcher
+		pdbWatcher = PDBWatcher{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "PDBWatcher",
+				APIVersion: "apps.mydomain.com/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pdb.Name,
+				Namespace: pdb.Namespace,
+			},
+			Spec: PDBWatcherSpec{
+				Name: pdb.Name,
```

Comment:
want to add target ref here to the deployment?

Reply:
should be able to use discover deployment or optionally an annotation your pdb reconiler leaves behind

Reply:
https://github.com/paulgmiller/k8s-pdb-autoscaler/pull/20